sphinx/ext/apidoc.py

```diff
@@ -53,6 +55,26 @@
 template_dir = path.join(package_dir, 'templates', 'apidoc')
 
 
+def _render_template(
+    template_path: Sequence[str],
+    template_basename: str,
+    context: dict[str, Any],
+) -> str:
+    """Attempt to render ReST output using a provided template search path
+    and context.
+
+    Priority is given to template files with a .jinja suffix.
+    """
+    for template_dir in template_path:
```

Comment:
Why not pass the `template_path` to `ReSTRenderer`?

Reply:
That would make for simpler code, but it would also mean that we would want to attempt to render the `_t` suffix templates before `.jinja` -- because otherwise, even if the operator specified a custom `--templatedir`, then the built-in `.jinja` templates within the Sphinx codebase would always be found first when `ReSTRenderer` searches the template paths.

That's probably OK - the only situation where I can imagine it would cause problems would be if sites have both `_t` and `.jinja` templates, and that shouldn't happen.  But it is inconsistent with other parts of the code where `.jinja` rendering is attempted first.

Reply:
...I suppose an alternative could be to push the `.jinja`/`_t` template detection logic down into the `ReSTRenderer` class itself.  I was reluctant to do that due to the fact that it might cause side-effects for other users of that class, but perhaps it'd be worthwhile to keep the interface straightforward.

Reply:
I've prepared a possible refactor for this, and will push that soon.  It could also allow streamlining some [similar template-lookup code in the `imgmath` extension](https://github.com/sphinx-doc/sphinx/blob/341b0417d5ccc2b495276cd60c5c6d8accd2a72c/sphinx/ext/imgmath.py#L111-L114) -- but one thing at a time.